tests/test_base.py

```diff
@@ -46,6 +50,35 @@ def test_base_delete_connection_error():
         j = n.job.deregister_job("example")
 
 
+@mock.patch("nomad.api.base.requests.Session")
+def test_base_raise_exception_not_requests_response_object(mock_requests):
+    mock_requests().delete.side_effect = [requests.RequestException()]
+
+    try:
+        n = nomad.Nomad(
+            host="162.16.10.102",
+            port=common.NOMAD_PORT,
+            timeout=0.001,
+            verify=False
+        )
+
+        _ = n.job.deregister_job("example")
+
+    except nomad.api.exceptions.BaseNomadException as err:
```

Comment:
[assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) as a context manager ends up being a clearer way of handling this type of test case imho.

Reply:
Completely agree with your points, I don't know why I decided to not uses pytest.raises context manager (since ive used it elsewhere), I'll swap that out!